### PR TITLE
Add e2e test for proxy config

### PR DESCRIPTION
**Pull request by abhinavmpandey08**
*Issue #, if available:*

*Description of changes:*
Adding e2e tests for testing proxy on ubuntu clusters

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/297#" title="Author self-approved">abhinavmpandey08</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [abhinavmpandey08]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: cmd/integration_test/build/buildspecs/test-eks-a-cli.yml
Comment (pokearu) [inline cmd/integration_test/build/buildspecs/test-eks-a-cli.yml:7]: Do we need some extra VPN logic for this to work in our test environment?

**Comment by abhay-krishna** [comment]
@abhinavmpandey08: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-attribution-files-presubmit | 06a2eb63c4797e69be1551bfc6202cfa043e7ef4 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/297/eks-anywhere-attribution-files-presubmit/1441123976926466048) | `/test eks-anywhere-attribution-files-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=297). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:abhinavmpandey08).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: cmd/integration_test/build/buildspecs/test-eks-a-cli.yml
Comment (abhinavmpandey08) [reply cmd/integration_test/build/buildspecs/test-eks-a-cli.yml:7]: we will need to add vpn routes to the proper subnets. I will create a cr for it soon

File: test/framework/proxy.go
Comment (vivek-koppuru) [inline test/framework/proxy.go:11]: Can we add these requirements to the README?

File: test/framework/vsphere.go
Comment (vivek-koppuru) [inline test/framework/vsphere.go:175]: Any reason we want to separate the env vars here vs reusing the network env var for the private use case? Is it because we only define each of the env vars once and we need a way to separate the configuration?

File: test/framework/vsphere.go
Comment (vivek-koppuru) [inline test/framework/vsphere.go:113]: Do we need to call validate here?

File: test/framework/vsphere.go
Comment (vivek-koppuru) [inline test/framework/vsphere.go:180]: Also validate here?

Comment (abhinavmpandey08) [reply test/framework/vsphere.go:180]: This is validated inside the `NewVSphere`

File: test/framework/proxy.go
Comment (abhinavmpandey08) [reply test/framework/proxy.go:11]: done

File: test/framework/vsphere.go
Comment (vivek-koppuru) [reply test/framework/vsphere.go:114]: Got it I missed that

**Review by vivek-koppuru — APPROVED** [review]
/lgtm

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/297#" title="Author self-approved">abhinavmpandey08</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/297#pullrequestreview-766012760" title="LGTM">vivek-koppuru</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [abhinavmpandey08,vivek-koppuru]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
